### PR TITLE
profiler: add WithSite option and DD_SITE environment variable.

### PR DESCRIPTION
**Pull request by knusbaum**
WithSite and DD_SITE will determine the apiURL that profiles are sent to.

I'm not sure if `WithURL` should be deprecated. Looking for feedback on this approach to implementing `DD_SITE` given we have `WithURL`.

### CODE REVIEW

File: profiler/options.go [hunk 1]
```diff
@@ -52,6 +52,10 @@ type config struct {
 	blockRate     int
 }
 
+func profilingURLForSite(site string) string {
```
Comment (gbbr) [inline profiler/options.go:55]:
```suggestion
func urlForSite(site string) string {
```
Will it be enough? I think the "profiling" context is clear from the package...

Also, should we do some validation here to make sure we catch any potential input errors from the user?

Comment (knusbaum) [reply profiler/options.go:55]: 👍 Sure.

Regarding validation, how strict do you think we want to be? Should we just have a whitelist of sites we compare against? Should we provide constants?

Comment (gbbr) [reply profiler/options.go:55]: I think it should suffice to ensure that the final URL compiles using [`url.Parse`](https://golang.org/pkg/net/url/#Parse) ?

Comment (knusbaum) [reply profiler/options.go:55]: 👍 That works for me. Good idea.

File: profiler/options.go [hunk 2]
```diff
@@ -52,6 +55,14 @@ type config struct {
 	blockRate     int
 }
 
+func urlForSite(site string) (string, error) {
+	u := fmt.Sprintf("https://intake.profile.%s/v1/input", site)
+	if _, err := url.Parse(u); err != nil {
+		return "", err
+	}
+	return u, nil
```
Comment (gbbr) [inline profiler/options.go:63]: nit
```suggestion
_, err := url.Parse(u)
return u, err
```

File: profiler/options.go [hunk 3]
```diff
@@ -60,9 +71,10 @@ func (c *config) addProfileType(t ProfileType) {
 }
 
 func defaultConfig() *config {
+	u, _ := urlForSite(defaultSite)
```
Comment (gbbr) [inline profiler/options.go:74]: I'm terrified by errors using `_` and even though this is legit, we never know who's going to change the argument and let the error ignored. This also means that the error is pointless, or that we would have no use for it here. I would suggest instead to restore `defaultAPIURL` and simply use that here. Just like it was before. `defaultSite` is not really needed anyway, right?

File: profiler/options.go [hunk 4]
```diff
@@ -174,3 +189,16 @@ func WithStatsd(client StatsdClient) Option {
...
 	}
 }
+
+// WithSite specifies the datadog site (datadoghq.com, datadoghq.eu, etc.)
+// which profiles will be sent to.
+func WithSite(site string) Option {
+	return func(cfg *config) {
+		u, err := urlForSite(site)
+		if err != nil {
+			log.Warn("profiler: ignoring invalid site %s: %s", site, err)
```
Comment (gbbr) [inline profiler/options.go:199]: Maybe we should say something like `invalid site provided, using https://intake.profile.datadoghq.com/v1/input`. I would even call this a fatal error but we don't want to ever crash the main program. Should this be an `Error` instead? 🤔

File: profiler/options_test.go [hunk 5]
```diff
@@ -62,6 +62,20 @@ func TestOptions(t *testing.T) {
 		assert.Equal(t, "serviceName", cfg.service)
 	})
 
+	t.Run("WithSite", func(t *testing.T) {
+		var cfg config
+		WithSite("some.other.site")(&cfg)
+		assert.Equal(t, "https://intake.profile.some.other.site/v1/input", cfg.apiURL)
+	})
+
+	t.Run("WithSite/override", func(t *testing.T) {
```
Comment (gbbr) [inline profiler/options_test.go:71]: Fake subtest? 😄

File: profiler/options.go (hunk 2 above)
Comment (knusbaum) [reply profiler/options.go:63]: Whoops, this is silly!

File: profiler/options.go (hunk 4 above)
Comment (knusbaum) [reply profiler/options.go:199]: Yes, we should probably log this as an `Error`. I agree, ideally this would be a `Fatal` error, but obviously can't be. This is where we should do something like #540 for `profiler.Start()`, to be able to report configuration or startup errors.

File: profiler/options.go (hunk 3 above)
Comment (knusbaum) [reply profiler/options.go:74]: Sure, I think this makes sense.

File: profiler/options_test.go (hunk 5 above)
Comment (knusbaum) [reply profiler/options_test.go:71]: Sorry, I don't follow. This subtest makes sure that `WithSite` will override `DD_SITE`

Comment (gbbr) [reply profiler/options_test.go:71]: I was just teasing. It's all good here. When you write sub-tests they get displayed like this. So this will look as if you have a test called `WithSite` and a subtest called `override`, but that's not true.

Comment (knusbaum) [reply profiler/options_test.go:71]: I see, I see. That's how I did the others as well. :)
Maybe in the future I should stick with `-` in subtest names to avoid confusion.

Comment (gbbr) [reply profiler/options_test.go:71]: This is all great. No need to change anything.

File: profiler/options_test.go [hunk 6]
```diff
@@ -147,7 +168,7 @@ func TestDefaultConfig(t *testing.T) {
 	t.Run("base", func(t *testing.T) {
 		cfg := defaultConfig()
 		assert := assert.New(t)
-		assert.Equal(defaultAPIURL, cfg.apiURL)
+		assert.Equal("https://intake.profile.datadoghq.com/v1/input", cfg.apiURL)
```
Comment (gbbr) [inline profiler/options_test.go:171]: You can undo this change too, I think...

File: profiler/options_test.go [hunk 7]
```diff
@@ -62,6 +62,20 @@ func TestOptions(t *testing.T) {
 		assert.Equal(t, "serviceName", cfg.service)
 	})
 
+	t.Run("WithSite", func(t *testing.T) {
+		var cfg config
+		WithSite("some.other.site")(&cfg)
```
Comment (gbbr) [inline profiler/options_test.go:67]: I will echo @palazzem's usual comments here: let's use a real example, like `datadoghq.eu`.

File: profiler/profiler_test.go [hunk 8]
```diff
@@ -36,7 +36,6 @@ func TestStart(t *testing.T) {
 		if host, err := os.Hostname(); err != nil {
 			assert.Equal(host, activeProfiler.cfg.hostname)
 		}
-		assert.Equal(defaultAPIURL, activeProfiler.cfg.apiURL)
```
Comment (gbbr) [inline profiler/profiler_test.go:39]: Why did you remove this? Accident?

Comment (knusbaum) [reply profiler/profiler_test.go:39]: Not sure how i managed to make that mistake...
